From 57c566fae9aa1e74a48f7272b14590bae9f7a00a Mon Sep 17 00:00:00 2001
From: Christian Hergert <chergert@redhat.com>
Date: Mon, 25 Mar 2019 17:04:27 -0700
Subject: [PATCH] build: fix gresource usage with meson 0.50
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This will probably be fixed upstream¹, but we can easily work
around it for now. Fixes build with Meson 0.50.0

¹ https://github.com/mesonbuild/meson/issues/5049
---
 src/gstyle/meson.build                        | 3 +--
 src/libide/code/meson.build                   | 2 +-
 src/libide/editor/meson.build                 | 2 +-
 src/libide/greeter/meson.build                | 2 +-
 src/libide/gui/meson.build                    | 2 +-
 src/libide/sourceview/meson.build             | 2 +-
 src/libide/terminal/meson.build               | 2 +-
 src/libide/webkit/meson.build                 | 2 +-
 src/plugins/auto-save/meson.build             | 2 +-
 src/plugins/autotools/meson.build             | 2 +-
 src/plugins/beautifier/meson.build            | 2 +-
 src/plugins/buffer-monitor/meson.build        | 2 +-
 src/plugins/buildconfig/meson.build           | 2 +-
 src/plugins/buildsystem/meson.build           | 2 +-
 src/plugins/buildui/meson.build               | 2 +-
 src/plugins/c-pack/meson.build                | 2 +-
 src/plugins/clang/meson.build                 | 2 +-
 src/plugins/cmake/meson.build                 | 2 +-
 src/plugins/code-index/meson.build            | 2 +-
 src/plugins/codeui/meson.build                | 2 +-
 src/plugins/color-picker/meson.build          | 2 +-
 src/plugins/command-bar/meson.build           | 2 +-
 src/plugins/comment-code/meson.build          | 2 +-
 src/plugins/create-project/meson.build        | 2 +-
 src/plugins/ctags/meson.build                 | 2 +-
 src/plugins/debuggerui/meson.build            | 2 +-
 src/plugins/devhelp/meson.build               | 2 +-
 src/plugins/deviced/meson.build               | 2 +-
 src/plugins/deviceui/meson.build              | 2 +-
 src/plugins/doap/meson.build                  | 2 +-
 src/plugins/editor/meson.build                | 2 +-
 src/plugins/editorconfig/meson.build          | 2 +-
 src/plugins/emacs/meson.build                 | 2 +-
 src/plugins/file-search/meson.build           | 2 +-
 src/plugins/flatpak/meson.build               | 2 +-
 src/plugins/gcc/meson.build                   | 2 +-
 src/plugins/gdb/meson.build                   | 2 +-
 src/plugins/gettext/meson.build               | 2 +-
 src/plugins/git/meson.build                   | 2 +-
 src/plugins/glade/meson.build                 | 2 +-
 src/plugins/gnome-code-assistance/meson.build | 2 +-
 src/plugins/greeter/meson.build               | 2 +-
 src/plugins/grep/meson.build                  | 2 +-
 src/plugins/history/meson.build               | 2 +-
 src/plugins/html-completion/meson.build       | 2 +-
 src/plugins/line-spacing/meson.build          | 2 +-
 src/plugins/ls/meson.build                    | 2 +-
 src/plugins/meson/meson.build                 | 2 +-
 src/plugins/messages/meson.build              | 2 +-
 src/plugins/modelines/meson.build             | 2 +-
 src/plugins/newcomers/meson.build             | 2 +-
 src/plugins/notification/meson.build          | 2 +-
 src/plugins/omni-gutter/meson.build           | 2 +-
 src/plugins/podman/meson.build                | 2 +-
 src/plugins/project-tree/meson.build          | 2 +-
 src/plugins/python-pack/meson.build           | 2 +-
 src/plugins/qemu/meson.build                  | 2 +-
 src/plugins/quick-highlight/meson.build       | 2 +-
 src/plugins/recent/meson.build                | 2 +-
 src/plugins/restore-cursor/meson.build        | 2 +-
 src/plugins/retab/meson.build                 | 2 +-
 src/plugins/snippets/meson.build              | 2 +-
 src/plugins/spellcheck/meson.build            | 2 +-
 src/plugins/sublime/meson.build               | 2 +-
 src/plugins/support/meson.build               | 2 +-
 src/plugins/symbol-tree/meson.build           | 2 +-
 src/plugins/sysprof/meson.build               | 2 +-
 src/plugins/sysroot/meson.build               | 2 +-
 src/plugins/terminal/meson.build              | 2 +-
 src/plugins/testui/meson.build                | 2 +-
 src/plugins/todo/meson.build                  | 2 +-
 src/plugins/trim-spaces/meson.build           | 2 +-
 src/plugins/vcsui/meson.build                 | 2 +-
 src/plugins/vim/meson.build                   | 2 +-
 src/plugins/words/meson.build                 | 2 +-
 src/plugins/xml-pack/meson.build              | 2 +-
 76 files changed, 76 insertions(+), 77 deletions(-)

diff --git a/src/gstyle/meson.build b/src/gstyle/meson.build
index 4d8c310fb..fb3247996 100644
--- a/src/gstyle/meson.build
+++ b/src/gstyle/meson.build
@@ -65,8 +65,7 @@ libgstyle_sources = [
   'gstyle-slidein.c',
   'gstyle-utils.c',
   'gstyle-xyz.c',
-  libgstyle_resources[0],
-  libgstyle_resources[1],
+  libgstyle_resources,
 ]
 
 libgstyle_deps = [
diff --git a/src/libide/code/meson.build b/src/libide/code/meson.build
index 14e7979b2..ddacdc162 100644
--- a/src/libide/code/meson.build
+++ b/src/libide/code/meson.build
@@ -138,7 +138,7 @@ libide_code_resources = gnome.compile_resources(
   c_name: 'ide_code',
 )
 libide_code_generated_headers += [libide_code_resources[1]]
-libide_code_generated_sources += libide_code_resources[0]
+libide_code_generated_sources += libide_code_resources
 
 
 #
diff --git a/src/libide/editor/meson.build b/src/libide/editor/meson.build
index ceaf83d3a..2838425b7 100644
--- a/src/libide/editor/meson.build
+++ b/src/libide/editor/meson.build
@@ -71,7 +71,7 @@ libide_editor_resources = gnome.compile_resources(
   c_name: 'ide_editor',
 )
 libide_editor_generated_headers += [libide_editor_resources[1]]
-libide_editor_sources += libide_editor_resources[0]
+libide_editor_sources += libide_editor_resources
 
 #
 # Dependencies
diff --git a/src/libide/greeter/meson.build b/src/libide/greeter/meson.build
index 2baa43503..3968ca41e 100644
--- a/src/libide/greeter/meson.build
+++ b/src/libide/greeter/meson.build
@@ -51,7 +51,7 @@ libide_greeter_resources = gnome.compile_resources(
   c_name: 'ide_greeter',
 )
 libide_greeter_generated_headers += [libide_greeter_resources[1]]
-libide_greeter_private_sources += libide_greeter_resources[0]
+libide_greeter_private_sources += libide_greeter_resources
 
 
 #
diff --git a/src/libide/gui/meson.build b/src/libide/gui/meson.build
index 3c494df9a..364d9cb12 100644
--- a/src/libide/gui/meson.build
+++ b/src/libide/gui/meson.build
@@ -163,7 +163,7 @@ libide_gui_resources = gnome.compile_resources(
   c_name: 'ide_gui',
 )
 libide_gui_generated_headers += [libide_gui_resources[1]]
-libide_gui_sources += libide_gui_resources[0]
+libide_gui_sources += libide_gui_resources
 
 
 #
diff --git a/src/libide/sourceview/meson.build b/src/libide/sourceview/meson.build
index 7d07a0b0c..c4ba12d1f 100644
--- a/src/libide/sourceview/meson.build
+++ b/src/libide/sourceview/meson.build
@@ -106,7 +106,7 @@ libide_sourceview_resources = gnome.compile_resources(
   c_name: 'ide_sourceview',
 )
 libide_sourceview_generated_headers += [libide_sourceview_resources[1]]
-libide_sourceview_generated_sources += libide_sourceview_resources[0]
+libide_sourceview_generated_sources += libide_sourceview_resources
 
 #
 # Enum generation
diff --git a/src/libide/terminal/meson.build b/src/libide/terminal/meson.build
index 4ab3a67cd..c316afac9 100644
--- a/src/libide/terminal/meson.build
+++ b/src/libide/terminal/meson.build
@@ -76,7 +76,7 @@ libide_terminal_deps = [
 #
 
 libide_terminal = static_library('ide-terminal-' + libide_api_version,
-   libide_terminal_public_sources + libide_terminal_private_sources + [libide_terminal_resources[0]],
+   libide_terminal_public_sources + libide_terminal_private_sources + libide_terminal_resources,
    dependencies: libide_terminal_deps,
          c_args: libide_args + release_args + ['-DIDE_TERMINAL_COMPILATION'],
 )
diff --git a/src/libide/webkit/meson.build b/src/libide/webkit/meson.build
index d92786c26..e1767ae46 100644
--- a/src/libide/webkit/meson.build
+++ b/src/libide/webkit/meson.build
@@ -17,7 +17,7 @@ libide_webkit_resources = gnome.compile_resources(
   c_name: 'ide_webkit',
 )
 libide_webkit_generated_headers = [libide_webkit_resources[1]]
-libide_webkit_sources += libide_webkit_resources[0]
+libide_webkit_sources += libide_webkit_resources
 
 #
 # Dependencies
diff --git a/src/plugins/auto-save/meson.build b/src/plugins/auto-save/meson.build
index 8beb1b0ac..4c12716f0 100644
--- a/src/plugins/auto-save/meson.build
+++ b/src/plugins/auto-save/meson.build
@@ -9,4 +9,4 @@ plugin_auto_save_resources = gnome.compile_resources(
   c_name: 'gbp_auto_save',
 )
 
-plugins_sources += plugin_auto_save_resources[0]
+plugins_sources += plugin_auto_save_resources
diff --git a/src/plugins/autotools/meson.build b/src/plugins/autotools/meson.build
index 93c592742..b9ab596af 100644
--- a/src/plugins/autotools/meson.build
+++ b/src/plugins/autotools/meson.build
@@ -20,6 +20,6 @@ plugin_autotools_resources = gnome.compile_resources(
   c_name: 'gbp_autotools',
 )
 
-plugins_sources += plugin_autotools_resources[0]
+plugins_sources += plugin_autotools_resources
 
 endif
diff --git a/src/plugins/beautifier/meson.build b/src/plugins/beautifier/meson.build
index 268957d2d..af37614f2 100644
--- a/src/plugins/beautifier/meson.build
+++ b/src/plugins/beautifier/meson.build
@@ -14,6 +14,6 @@ plugin_beautifier_resources = gnome.compile_resources(
   c_name: 'gbp_beautifier',
 )
 
-plugins_sources += plugin_beautifier_resources[0]
+plugins_sources += plugin_beautifier_resources
 
 endif
diff --git a/src/plugins/buffer-monitor/meson.build b/src/plugins/buffer-monitor/meson.build
index e4669de4c..5df90d1e6 100644
--- a/src/plugins/buffer-monitor/meson.build
+++ b/src/plugins/buffer-monitor/meson.build
@@ -9,4 +9,4 @@ plugin_buffer_monitor_resources = gnome.compile_resources(
   c_name: 'gbp_buffer_monitor',
 )
 
-plugins_sources += plugin_buffer_monitor_resources[0]
+plugins_sources += plugin_buffer_monitor_resources
diff --git a/src/plugins/buildconfig/meson.build b/src/plugins/buildconfig/meson.build
index 528ba7656..ca9d20d75 100644
--- a/src/plugins/buildconfig/meson.build
+++ b/src/plugins/buildconfig/meson.build
@@ -11,4 +11,4 @@ plugin_buildconfig_resources = gnome.compile_resources(
   c_name: 'gbp_buildconfig',
 )
 
-plugins_sources += plugin_buildconfig_resources[0]
+plugins_sources += plugin_buildconfig_resources
diff --git a/src/plugins/buildsystem/meson.build b/src/plugins/buildsystem/meson.build
index 2de1e508c..acbca0606 100644
--- a/src/plugins/buildsystem/meson.build
+++ b/src/plugins/buildsystem/meson.build
@@ -9,4 +9,4 @@ plugin_buildsystem_resources = gnome.compile_resources(
   c_name: 'gbp_buildsystem',
 )
 
-plugins_sources += plugin_buildsystem_resources[0]
+plugins_sources += plugin_buildsystem_resources
diff --git a/src/plugins/buildui/meson.build b/src/plugins/buildui/meson.build
index eb2e71947..6524ce012 100644
--- a/src/plugins/buildui/meson.build
+++ b/src/plugins/buildui/meson.build
@@ -19,4 +19,4 @@ plugin_buildui_resources = gnome.compile_resources(
   c_name: 'gbp_buildui',
 )
 
-plugins_sources += plugin_buildui_resources[0]
+plugins_sources += plugin_buildui_resources
diff --git a/src/plugins/c-pack/meson.build b/src/plugins/c-pack/meson.build
index a1f60fc9f..0fffd5f9f 100644
--- a/src/plugins/c-pack/meson.build
+++ b/src/plugins/c-pack/meson.build
@@ -17,7 +17,7 @@ plugin_c_pack_resources = gnome.compile_resources(
   c_name: 'ide_c',
 )
 
-plugins_sources += plugin_c_pack_resources[0]
+plugins_sources += plugin_c_pack_resources
 
 test_cpack = executable('test-cpack',
   'test-cpack.c', 'c-parse-helper.c',
diff --git a/src/plugins/clang/meson.build b/src/plugins/clang/meson.build
index 01381b74f..3a7b87f0b 100644
--- a/src/plugins/clang/meson.build
+++ b/src/plugins/clang/meson.build
@@ -41,7 +41,7 @@ plugin_clang_resources = gnome.compile_resources(
   c_name: 'gbp_clang',
 )
 
-plugins_sources += plugin_clang_resources[0]
+plugins_sources += plugin_clang_resources
 
 add_languages('cpp') # Needed for llvm dep
 llvm_dep = dependency('llvm', version: '>= 3.5')
diff --git a/src/plugins/cmake/meson.build b/src/plugins/cmake/meson.build
index 37abefb9b..f97154023 100644
--- a/src/plugins/cmake/meson.build
+++ b/src/plugins/cmake/meson.build
@@ -17,6 +17,6 @@ plugin_cmake_resources = gnome.compile_resources(
   c_name: 'gbp_cmake',
 )
 
-plugins_sources += plugin_cmake_resources[0]
+plugins_sources += plugin_cmake_resources
 
 endif
diff --git a/src/plugins/code-index/meson.build b/src/plugins/code-index/meson.build
index c6f175d81..2b9fd4819 100644
--- a/src/plugins/code-index/meson.build
+++ b/src/plugins/code-index/meson.build
@@ -21,6 +21,6 @@ plugin_code_index_resources = gnome.compile_resources(
   c_name: 'gbp_code_index',
 )
 
-plugins_sources += plugin_code_index_resources[0]
+plugins_sources += plugin_code_index_resources
 
 endif
diff --git a/src/plugins/codeui/meson.build b/src/plugins/codeui/meson.build
index 751517423..928a34315 100644
--- a/src/plugins/codeui/meson.build
+++ b/src/plugins/codeui/meson.build
@@ -10,4 +10,4 @@ plugin_codeui_resources = gnome.compile_resources(
   c_name: 'gbp_codeui',
 )
 
-plugins_sources += plugin_codeui_resources[0]
+plugins_sources += plugin_codeui_resources
diff --git a/src/plugins/color-picker/meson.build b/src/plugins/color-picker/meson.build
index cbfca15eb..cdd68d28b 100644
--- a/src/plugins/color-picker/meson.build
+++ b/src/plugins/color-picker/meson.build
@@ -22,6 +22,6 @@ plugin_color_picker_resources = gnome.compile_resources(
   c_name: 'gbp_color_picker',
 )
 
-plugins_sources += plugin_color_picker_resources[0]
+plugins_sources += plugin_color_picker_resources
 
 endif
diff --git a/src/plugins/command-bar/meson.build b/src/plugins/command-bar/meson.build
index 76bde4928..aebde636f 100644
--- a/src/plugins/command-bar/meson.build
+++ b/src/plugins/command-bar/meson.build
@@ -15,4 +15,4 @@ plugin_command_bar_resources = gnome.compile_resources(
   c_name: 'gbp_command_bar',
 )
 
-plugins_sources += plugin_command_bar_resources[0]
+plugins_sources += plugin_command_bar_resources
diff --git a/src/plugins/comment-code/meson.build b/src/plugins/comment-code/meson.build
index 703df526c..e0eef1e0d 100644
--- a/src/plugins/comment-code/meson.build
+++ b/src/plugins/comment-code/meson.build
@@ -9,4 +9,4 @@ plugin_comment_code_resources = gnome.compile_resources(
   c_name: 'gbp_comment_code',
 )
 
-plugins_sources += plugin_comment_code_resources[0]
+plugins_sources += plugin_comment_code_resources
diff --git a/src/plugins/create-project/meson.build b/src/plugins/create-project/meson.build
index e99c1fc80..94f1f09b4 100644
--- a/src/plugins/create-project/meson.build
+++ b/src/plugins/create-project/meson.build
@@ -12,4 +12,4 @@ plugin_create_project_resources = gnome.compile_resources(
   c_name: 'gbp_create_project',
 )
 
-plugins_sources += plugin_create_project_resources[0]
+plugins_sources += plugin_create_project_resources
diff --git a/src/plugins/ctags/meson.build b/src/plugins/ctags/meson.build
index acd7fa79a..36c8da888 100644
--- a/src/plugins/ctags/meson.build
+++ b/src/plugins/ctags/meson.build
@@ -24,7 +24,7 @@ plugin_ctags_resources = gnome.compile_resources(
   c_name: 'gbp_ctags',
 )
 
-plugins_sources += plugin_ctags_resources[0]
+plugins_sources += plugin_ctags_resources
 
 test_ctags = executable('test-ctags',
   'test-ctags.c', 'ide-ctags-index.c',
diff --git a/src/plugins/debuggerui/meson.build b/src/plugins/debuggerui/meson.build
index cd77cec81..87d981288 100644
--- a/src/plugins/debuggerui/meson.build
+++ b/src/plugins/debuggerui/meson.build
@@ -18,4 +18,4 @@ plugin_debuggerui_resources = gnome.compile_resources(
   c_name: 'gbp_debuggerui',
 )
 
-plugins_sources += plugin_debuggerui_resources[0]
+plugins_sources += plugin_debuggerui_resources
diff --git a/src/plugins/devhelp/meson.build b/src/plugins/devhelp/meson.build
index 685f73f51..e4cbc0223 100644
--- a/src/plugins/devhelp/meson.build
+++ b/src/plugins/devhelp/meson.build
@@ -20,6 +20,6 @@ plugin_devhelp_resources = gnome.compile_resources(
   c_name: 'gbp_devhelp',
 )
 
-plugins_sources += plugin_devhelp_resources[0]
+plugins_sources += plugin_devhelp_resources
 
 endif
diff --git a/src/plugins/deviced/meson.build b/src/plugins/deviced/meson.build
index c728cb5d2..3c2453331 100644
--- a/src/plugins/deviced/meson.build
+++ b/src/plugins/deviced/meson.build
@@ -21,6 +21,6 @@ plugins_deps += [
   dependency('libdeviced', version: '>=3.27.4'),
 ]
 
-plugins_sources += plugin_deviced_resources[0]
+plugins_sources += plugin_deviced_resources
 
 endif
diff --git a/src/plugins/deviceui/meson.build b/src/plugins/deviceui/meson.build
index d1d00235c..b9b125ef5 100644
--- a/src/plugins/deviceui/meson.build
+++ b/src/plugins/deviceui/meson.build
@@ -9,4 +9,4 @@ plugin_deviceui_resources = gnome.compile_resources(
   c_name: 'gbp_deviceui',
 )
 
-plugins_sources += plugin_deviceui_resources[0]
+plugins_sources += plugin_deviceui_resources
diff --git a/src/plugins/doap/meson.build b/src/plugins/doap/meson.build
index ebd2ffb74..f302c2520 100644
--- a/src/plugins/doap/meson.build
+++ b/src/plugins/doap/meson.build
@@ -9,4 +9,4 @@ plugin_doap_resources = gnome.compile_resources(
   c_name: 'gbp_doap',
 )
 
-plugins_sources += plugin_doap_resources[0]
+plugins_sources += plugin_doap_resources
diff --git a/src/plugins/editor/meson.build b/src/plugins/editor/meson.build
index 7c0101be7..08655c113 100644
--- a/src/plugins/editor/meson.build
+++ b/src/plugins/editor/meson.build
@@ -15,4 +15,4 @@ plugin_editor_resources = gnome.compile_resources(
   c_name: 'gbp_editor',
 )
 
-plugins_sources += plugin_editor_resources[0]
+plugins_sources += plugin_editor_resources
diff --git a/src/plugins/editorconfig/meson.build b/src/plugins/editorconfig/meson.build
index 8cbf269d1..2637ff5de 100644
--- a/src/plugins/editorconfig/meson.build
+++ b/src/plugins/editorconfig/meson.build
@@ -14,7 +14,7 @@ plugin_editorconfig_resources = gnome.compile_resources(
   c_name: 'gbp_editorconfig',
 )
 
-plugins_sources += plugin_editorconfig_resources[0]
+plugins_sources += plugin_editorconfig_resources
 plugins_deps += libeditorconfig_dep
 
 endif
diff --git a/src/plugins/emacs/meson.build b/src/plugins/emacs/meson.build
index a568a5666..bd5375286 100644
--- a/src/plugins/emacs/meson.build
+++ b/src/plugins/emacs/meson.build
@@ -9,4 +9,4 @@ plugin_emacs_resources = gnome.compile_resources(
   c_name: 'gbp_emacs',
 )
 
-plugins_sources += plugin_emacs_resources[0]
+plugins_sources += plugin_emacs_resources
diff --git a/src/plugins/file-search/meson.build b/src/plugins/file-search/meson.build
index 8ae908f90..eeb9406e5 100644
--- a/src/plugins/file-search/meson.build
+++ b/src/plugins/file-search/meson.build
@@ -13,6 +13,6 @@ plugin_file_search_resources = gnome.compile_resources(
   c_name: 'gbp_file_search',
 )
 
-plugins_sources += plugin_file_search_resources[0]
+plugins_sources += plugin_file_search_resources
 
 endif
diff --git a/src/plugins/flatpak/meson.build b/src/plugins/flatpak/meson.build
index 438dd08f2..6fab01c1e 100644
--- a/src/plugins/flatpak/meson.build
+++ b/src/plugins/flatpak/meson.build
@@ -40,6 +40,6 @@ plugins_deps += [
   dependency('libsoup-2.4', version: '>= 2.52.0'),
 ]
 
-plugins_sources += plugin_flatpak_resources[0]
+plugins_sources += plugin_flatpak_resources
 
 endif
diff --git a/src/plugins/gcc/meson.build b/src/plugins/gcc/meson.build
index 932bcdc92..cbd7520e3 100644
--- a/src/plugins/gcc/meson.build
+++ b/src/plugins/gcc/meson.build
@@ -12,4 +12,4 @@ plugin_gcc_resources = gnome.compile_resources(
   c_name: 'gbp_gcc',
 )
 
-plugins_sources += plugin_gcc_resources[0]
+plugins_sources += plugin_gcc_resources
diff --git a/src/plugins/gdb/meson.build b/src/plugins/gdb/meson.build
index 804213497..b24f2af68 100644
--- a/src/plugins/gdb/meson.build
+++ b/src/plugins/gdb/meson.build
@@ -11,7 +11,7 @@ plugin_gdb_resources = gnome.compile_resources(
   c_name: 'gbp_gdb',
 )
 
-plugins_sources += plugin_gdb_resources[0]
+plugins_sources += plugin_gdb_resources
 
 gdbwire = static_library('gdbwire', ['gdbwire.c'],
   c_args: [ '-Wno-redundant-decls',
diff --git a/src/plugins/gettext/meson.build b/src/plugins/gettext/meson.build
index 24ccf09b4..d3bbddb71 100644
--- a/src/plugins/gettext/meson.build
+++ b/src/plugins/gettext/meson.build
@@ -11,6 +11,6 @@ plugin_gettext_resources = gnome.compile_resources(
   c_name: 'gbp_gettext',
 )
 
-plugins_sources += plugin_gettext_resources[0]
+plugins_sources += plugin_gettext_resources
 
 endif
diff --git a/src/plugins/git/meson.build b/src/plugins/git/meson.build
index 79dc08181..894da68ac 100644
--- a/src/plugins/git/meson.build
+++ b/src/plugins/git/meson.build
@@ -25,6 +25,6 @@ plugin_git_resources = gnome.compile_resources(
   c_name: 'gbp_git',
 )
 
-plugins_sources += plugin_git_resources[0]
+plugins_sources += plugin_git_resources
 
 endif
diff --git a/src/plugins/glade/meson.build b/src/plugins/glade/meson.build
index c57630612..f365f9f54 100644
--- a/src/plugins/glade/meson.build
+++ b/src/plugins/glade/meson.build
@@ -21,6 +21,6 @@ plugins_deps += [
   dependency('gladeui-2.0', version: '>=3.22.0'),
 ]
 
-plugins_sources += plugin_glade_resources[0]
+plugins_sources += plugin_glade_resources
 
 endif
diff --git a/src/plugins/gnome-code-assistance/meson.build b/src/plugins/gnome-code-assistance/meson.build
index a7d86fa14..452dad6fd 100644
--- a/src/plugins/gnome-code-assistance/meson.build
+++ b/src/plugins/gnome-code-assistance/meson.build
@@ -18,6 +18,6 @@ plugin_gnome_code_assistance_resources = gnome.compile_resources(
   c_name: 'gbp_gnome_code_assistance',
 )
 
-plugins_sources += plugin_gnome_code_assistance_resources[0]
+plugins_sources += plugin_gnome_code_assistance_resources
 
 endif
diff --git a/src/plugins/greeter/meson.build b/src/plugins/greeter/meson.build
index 736f31520..a31c3ed9b 100644
--- a/src/plugins/greeter/meson.build
+++ b/src/plugins/greeter/meson.build
@@ -9,4 +9,4 @@ plugin_greeter_resources = gnome.compile_resources(
   c_name: 'gbp_greeter',
 )
 
-plugins_sources += plugin_greeter_resources[0]
+plugins_sources += plugin_greeter_resources
diff --git a/src/plugins/grep/meson.build b/src/plugins/grep/meson.build
index d6e586c7b..0803c6e91 100644
--- a/src/plugins/grep/meson.build
+++ b/src/plugins/grep/meson.build
@@ -14,6 +14,6 @@ plugin_grep_resources = gnome.compile_resources(
   c_name: 'gbp_grep',
 )
 
-plugins_sources += plugin_grep_resources[0]
+plugins_sources += plugin_grep_resources
 
 endif
diff --git a/src/plugins/history/meson.build b/src/plugins/history/meson.build
index 78d5e5cc1..7a327f5d2 100644
--- a/src/plugins/history/meson.build
+++ b/src/plugins/history/meson.build
@@ -11,4 +11,4 @@ plugin_history_resources = gnome.compile_resources(
   c_name: 'gbp_history',
 )
 
-plugins_sources += plugin_history_resources[0]
+plugins_sources += plugin_history_resources
diff --git a/src/plugins/html-completion/meson.build b/src/plugins/html-completion/meson.build
index a44ec62f4..556a9e48d 100644
--- a/src/plugins/html-completion/meson.build
+++ b/src/plugins/html-completion/meson.build
@@ -13,6 +13,6 @@ plugin_html_completion_resources = gnome.compile_resources(
   c_name: 'gbp_html_completion',
 )
 
-plugins_sources += plugin_html_completion_resources[0]
+plugins_sources += plugin_html_completion_resources
 
 endif
diff --git a/src/plugins/line-spacing/meson.build b/src/plugins/line-spacing/meson.build
index 3a6dc5b35..a7582797d 100644
--- a/src/plugins/line-spacing/meson.build
+++ b/src/plugins/line-spacing/meson.build
@@ -9,4 +9,4 @@ plugin_line_spacing_resources = gnome.compile_resources(
   c_name: 'gbp_line_spacing',
 )
 
-plugins_sources += plugin_line_spacing_resources[0]
+plugins_sources += plugin_line_spacing_resources
diff --git a/src/plugins/ls/meson.build b/src/plugins/ls/meson.build
index 99633be7f..27f4bb85e 100644
--- a/src/plugins/ls/meson.build
+++ b/src/plugins/ls/meson.build
@@ -11,4 +11,4 @@ plugin_ls_resources = gnome.compile_resources(
   c_name: 'gbp_ls',
 )
 
-plugins_sources += plugin_ls_resources[0]
+plugins_sources += plugin_ls_resources
diff --git a/src/plugins/meson/meson.build b/src/plugins/meson/meson.build
index f161fb0c8..4e16a7da5 100644
--- a/src/plugins/meson/meson.build
+++ b/src/plugins/meson/meson.build
@@ -25,6 +25,6 @@ plugin_meson_resources = gnome.compile_resources(
   c_name: 'gbp_meson',
 )
 
-plugins_sources += plugin_meson_resources[0]
+plugins_sources += plugin_meson_resources
 
 endif
diff --git a/src/plugins/messages/meson.build b/src/plugins/messages/meson.build
index af8a212fa..78b97b43f 100644
--- a/src/plugins/messages/meson.build
+++ b/src/plugins/messages/meson.build
@@ -10,4 +10,4 @@ plugin_messages_resources = gnome.compile_resources(
   c_name: 'gbp_messages',
 )
 
-plugins_sources += plugin_messages_resources[0]
+plugins_sources += plugin_messages_resources
diff --git a/src/plugins/modelines/meson.build b/src/plugins/modelines/meson.build
index 7916fc4ae..2306675af 100644
--- a/src/plugins/modelines/meson.build
+++ b/src/plugins/modelines/meson.build
@@ -12,6 +12,6 @@ plugin_modelines_resources = gnome.compile_resources(
   c_name: 'gbp_modelines',
 )
 
-plugins_sources += plugin_modelines_resources[0]
+plugins_sources += plugin_modelines_resources
 
 endif
diff --git a/src/plugins/newcomers/meson.build b/src/plugins/newcomers/meson.build
index be45e6b45..256d7a10c 100644
--- a/src/plugins/newcomers/meson.build
+++ b/src/plugins/newcomers/meson.build
@@ -12,6 +12,6 @@ plugin_newcomers_resources = gnome.compile_resources(
   c_name: 'gbp_newcomers',
 )
 
-plugins_sources += plugin_newcomers_resources[0]
+plugins_sources += plugin_newcomers_resources
 
 endif
diff --git a/src/plugins/notification/meson.build b/src/plugins/notification/meson.build
index e7d23dcd1..afe1260e2 100644
--- a/src/plugins/notification/meson.build
+++ b/src/plugins/notification/meson.build
@@ -11,6 +11,6 @@ plugin_notification_resources = gnome.compile_resources(
   c_name: 'gbp_notification',
 )
 
-plugins_sources += plugin_notification_resources[0]
+plugins_sources += plugin_notification_resources
 
 endif
diff --git a/src/plugins/omni-gutter/meson.build b/src/plugins/omni-gutter/meson.build
index 80f4168b8..36e8331e8 100644
--- a/src/plugins/omni-gutter/meson.build
+++ b/src/plugins/omni-gutter/meson.build
@@ -11,4 +11,4 @@ omni_gutter_resources = gnome.compile_resources(
   c_name: 'gbp_omni_gutter',
 )
 
-plugins_sources += omni_gutter_resources[0]
+plugins_sources += omni_gutter_resources
diff --git a/src/plugins/podman/meson.build b/src/plugins/podman/meson.build
index b1a589444..9fc50fdc4 100644
--- a/src/plugins/podman/meson.build
+++ b/src/plugins/podman/meson.build
@@ -13,6 +13,6 @@ plugin_podman_resources = gnome.compile_resources(
   c_name: 'gbp_podman'
 )
 
-plugins_sources += plugin_podman_resources[0]
+plugins_sources += plugin_podman_resources
 
 endif
diff --git a/src/plugins/project-tree/meson.build b/src/plugins/project-tree/meson.build
index c95aa92be..99f0e1f8c 100644
--- a/src/plugins/project-tree/meson.build
+++ b/src/plugins/project-tree/meson.build
@@ -15,4 +15,4 @@ plugin_project_tree_resources = gnome.compile_resources(
   c_name: 'gbp_project_tree',
 )
 
-plugins_sources += plugin_project_tree_resources[0]
+plugins_sources += plugin_project_tree_resources
diff --git a/src/plugins/python-pack/meson.build b/src/plugins/python-pack/meson.build
index a369bfa9f..96bd89d90 100644
--- a/src/plugins/python-pack/meson.build
+++ b/src/plugins/python-pack/meson.build
@@ -11,6 +11,6 @@ plugin_python_pack_resources = gnome.compile_resources(
   c_name: 'gbp_python_pack',
 )
 
-plugins_sources += plugin_python_pack_resources[0]
+plugins_sources += plugin_python_pack_resources
 
 endif
diff --git a/src/plugins/qemu/meson.build b/src/plugins/qemu/meson.build
index ca36d1bb7..32bcee7d8 100644
--- a/src/plugins/qemu/meson.build
+++ b/src/plugins/qemu/meson.build
@@ -11,6 +11,6 @@ plugin_qemu_resources = gnome.compile_resources(
   c_name: 'gbp_qemu',
 )
 
-plugins_sources += plugin_qemu_resources[0]
+plugins_sources += plugin_qemu_resources
 
 endif
diff --git a/src/plugins/quick-highlight/meson.build b/src/plugins/quick-highlight/meson.build
index 5bb496fb7..da0b06e7e 100644
--- a/src/plugins/quick-highlight/meson.build
+++ b/src/plugins/quick-highlight/meson.build
@@ -12,6 +12,6 @@ plugin_quick_highlight_resources = gnome.compile_resources(
   c_name: 'gbp_quick_highlight',
 )
 
-plugins_sources += plugin_quick_highlight_resources[0]
+plugins_sources += plugin_quick_highlight_resources
 
 endif
diff --git a/src/plugins/recent/meson.build b/src/plugins/recent/meson.build
index 9a3417060..6094decb0 100644
--- a/src/plugins/recent/meson.build
+++ b/src/plugins/recent/meson.build
@@ -10,4 +10,4 @@ plugin_recent_resources = gnome.compile_resources(
   c_name: 'gbp_recent',
 )
 
-plugins_sources += plugin_recent_resources[0]
+plugins_sources += plugin_recent_resources
diff --git a/src/plugins/restore-cursor/meson.build b/src/plugins/restore-cursor/meson.build
index 5ebbe8810..43d8f340f 100644
--- a/src/plugins/restore-cursor/meson.build
+++ b/src/plugins/restore-cursor/meson.build
@@ -9,4 +9,4 @@ plugin_restore_cursor_resources = gnome.compile_resources(
   c_name: 'gbp_restore_cursor',
 )
 
-plugins_sources += plugin_restore_cursor_resources[0]
+plugins_sources += plugin_restore_cursor_resources
diff --git a/src/plugins/retab/meson.build b/src/plugins/retab/meson.build
index ecb8e9476..c2c901d56 100644
--- a/src/plugins/retab/meson.build
+++ b/src/plugins/retab/meson.build
@@ -11,6 +11,6 @@ plugin_retab_resources = gnome.compile_resources(
   c_name: 'gbp_retab',
 )
 
-plugins_sources += plugin_retab_resources[0]
+plugins_sources += plugin_retab_resources
 
 endif
diff --git a/src/plugins/snippets/meson.build b/src/plugins/snippets/meson.build
index ab2e3fec2..007b4ecd7 100644
--- a/src/plugins/snippets/meson.build
+++ b/src/plugins/snippets/meson.build
@@ -12,4 +12,4 @@ snippets_resources = gnome.compile_resources(
   c_name: 'gbp_snippets',
 )
 
-plugins_sources += snippets_resources[0]
+plugins_sources += snippets_resources
diff --git a/src/plugins/spellcheck/meson.build b/src/plugins/spellcheck/meson.build
index 659d12621..c29516d0f 100644
--- a/src/plugins/spellcheck/meson.build
+++ b/src/plugins/spellcheck/meson.build
@@ -24,6 +24,6 @@ plugins_deps += [
   dependency('enchant-2'),
 ]
 
-plugins_sources += plugin_spellcheck_resources[0]
+plugins_sources += plugin_spellcheck_resources
 
 endif
diff --git a/src/plugins/sublime/meson.build b/src/plugins/sublime/meson.build
index 3272cfc5f..4ca6dfbbc 100644
--- a/src/plugins/sublime/meson.build
+++ b/src/plugins/sublime/meson.build
@@ -9,4 +9,4 @@ plugin_sublime_resources = gnome.compile_resources(
   c_name: 'gbp_sublime',
 )
 
-plugins_sources += plugin_sublime_resources[0]
+plugins_sources += plugin_sublime_resources
diff --git a/src/plugins/support/meson.build b/src/plugins/support/meson.build
index 23db282d9..9906f7eef 100644
--- a/src/plugins/support/meson.build
+++ b/src/plugins/support/meson.build
@@ -10,4 +10,4 @@ plugin_support_resources = gnome.compile_resources(
   c_name: 'gbp_support',
 )
 
-plugins_sources += plugin_support_resources[0]
+plugins_sources += plugin_support_resources
diff --git a/src/plugins/symbol-tree/meson.build b/src/plugins/symbol-tree/meson.build
index b40d3d4e6..1eeb61f54 100644
--- a/src/plugins/symbol-tree/meson.build
+++ b/src/plugins/symbol-tree/meson.build
@@ -12,4 +12,4 @@ plugin_symbol_tree_resources = gnome.compile_resources(
   c_name: 'gbp_symbol_tree',
 )
 
-plugins_sources += plugin_symbol_tree_resources[0]
+plugins_sources += plugin_symbol_tree_resources
diff --git a/src/plugins/sysprof/meson.build b/src/plugins/sysprof/meson.build
index 780e9ca9f..2e9eca880 100644
--- a/src/plugins/sysprof/meson.build
+++ b/src/plugins/sysprof/meson.build
@@ -19,6 +19,6 @@ plugin_sysprof_resources = gnome.compile_resources(
   c_name: 'gbp_sysprof',
 )
 
-plugins_sources += plugin_sysprof_resources[0]
+plugins_sources += plugin_sysprof_resources
 
 endif
diff --git a/src/plugins/sysroot/meson.build b/src/plugins/sysroot/meson.build
index aad3d6229..cb640d5cf 100644
--- a/src/plugins/sysroot/meson.build
+++ b/src/plugins/sysroot/meson.build
@@ -17,6 +17,6 @@ plugin_sysroot_resources = gnome.compile_resources(
   c_name: 'gbp_sysroot',
 )
 
-plugins_sources += plugin_sysroot_resources[0]
+plugins_sources += plugin_sysroot_resources
 
 endif
diff --git a/src/plugins/terminal/meson.build b/src/plugins/terminal/meson.build
index f13204f81..1c9dbc7e2 100644
--- a/src/plugins/terminal/meson.build
+++ b/src/plugins/terminal/meson.build
@@ -10,4 +10,4 @@ plugin_terminal_resources = gnome.compile_resources(
   c_name: 'gbp_terminal',
 )
 
-plugins_sources += plugin_terminal_resources[0]
+plugins_sources += plugin_terminal_resources
diff --git a/src/plugins/testui/meson.build b/src/plugins/testui/meson.build
index 019c44e67..3778eb403 100644
--- a/src/plugins/testui/meson.build
+++ b/src/plugins/testui/meson.build
@@ -11,4 +11,4 @@ plugin_testui_resources = gnome.compile_resources(
   c_name: 'gbp_testui',
 )
 
-plugins_sources += plugin_testui_resources[0]
+plugins_sources += plugin_testui_resources
diff --git a/src/plugins/todo/meson.build b/src/plugins/todo/meson.build
index 0e8b6f67a..91e789e0c 100644
--- a/src/plugins/todo/meson.build
+++ b/src/plugins/todo/meson.build
@@ -14,6 +14,6 @@ plugin_todo_resources = gnome.compile_resources(
   c_name: 'gbp_todo',
 )
 
-plugins_sources += plugin_todo_resources[0]
+plugins_sources += plugin_todo_resources
 
 endif
diff --git a/src/plugins/trim-spaces/meson.build b/src/plugins/trim-spaces/meson.build
index e366b0e69..a88635391 100644
--- a/src/plugins/trim-spaces/meson.build
+++ b/src/plugins/trim-spaces/meson.build
@@ -9,4 +9,4 @@ plugin_trim_spaces_resources = gnome.compile_resources(
   c_name: 'gbp_trim_spaces',
 )
 
-plugins_sources += plugin_trim_spaces_resources[0]
+plugins_sources += plugin_trim_spaces_resources
diff --git a/src/plugins/vcsui/meson.build b/src/plugins/vcsui/meson.build
index f85a51df3..35d44e02c 100644
--- a/src/plugins/vcsui/meson.build
+++ b/src/plugins/vcsui/meson.build
@@ -11,4 +11,4 @@ plugin_vcsui_resources = gnome.compile_resources(
   c_name: 'gbp_vcsui',
 )
 
-plugins_sources += plugin_vcsui_resources[0]
+plugins_sources += plugin_vcsui_resources
diff --git a/src/plugins/vim/meson.build b/src/plugins/vim/meson.build
index 8c20ea085..fcdcde818 100644
--- a/src/plugins/vim/meson.build
+++ b/src/plugins/vim/meson.build
@@ -12,4 +12,4 @@ plugin_vim_resources = gnome.compile_resources(
   c_name: 'gbp_vim',
 )
 
-plugins_sources += plugin_vim_resources[0]
+plugins_sources += plugin_vim_resources
diff --git a/src/plugins/words/meson.build b/src/plugins/words/meson.build
index 0d8cfdb15..9f81be059 100644
--- a/src/plugins/words/meson.build
+++ b/src/plugins/words/meson.build
@@ -13,6 +13,6 @@ plugin_words_resources = gnome.compile_resources(
   c_name: 'gbp_words',
 )
 
-plugins_sources += plugin_words_resources[0]
+plugins_sources += plugin_words_resources
 
 endif
diff --git a/src/plugins/xml-pack/meson.build b/src/plugins/xml-pack/meson.build
index 9338eae07..1137fdcaa 100644
--- a/src/plugins/xml-pack/meson.build
+++ b/src/plugins/xml-pack/meson.build
@@ -40,6 +40,6 @@ plugin_xml_pack_resources = gnome.compile_resources(
   c_name: 'gbp_xml_pack',
 )
 
-plugins_sources += plugin_xml_pack_resources[0]
+plugins_sources += plugin_xml_pack_resources
 
 endif
